### DISCUSSION TITLE
Using S2 as the core geometry

### DISCUSSION THREAD

**Discussion by nyurik** [maplibre/maplibre-tile-spec#9] [Ideas]
After speaking with @CraigglesO about [S2 geometries](https://s2geometry.io/), it seems like there might be some value in adapting S2 as the core tiling system due to its extended feature set - i.e. being able to cover the entire globe (with the polar caps), and also not distorting data as much as MVT/web mercator does.   Would it make sense in the context of COV tiles?

**Comment by mactrem** [comment]
I think support for s2 based tiling schemes is also tightly coupled to the map renderer, as it basically means there has to be support for rendering virtual globes. Currently the only open source map renderer I’m aware of which is performant in rendering virtual globes is Cesium. So only supporting S2 and not webmercator could be problematic. 

One approach could be to allow different tiling schemes like in the [Two Dimensional Tile Matrix Set and Tile Set Metadata](https://docs.ogc.org/is/17-083r4/17-083r4.html) specification of the OGC. I also like the approach of the [implicit tiling schema](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_implicit_tiling#subdivision-scheme) of the 3DTiles specification where you can use a quadtree or octree based subdivision schema in combination with [s2](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_bounding_volume_S2).

**Reply by nyurik** [reply]
:100: agree that both indexing schema should be supported. Actually, the indexing part of S2 is a problem for COM-tiles, whereas COV-tiles would not care at all about the indexing, but may (?) care about the geometry itself - i.e. my understanding is that S2 has a different approach to storing geometry than the 2D approach of MVT. So this was mostly a question on how the geometry rather than tile-indexing would be supported by COV-tiles (sigh, those names are really confusing sometimes :) )
